config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -416,6 +440,432 @@ type MetricsServerConfig struct {
+	// The value must be a valid filename.
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=255
+	QueryLogFile string `json:"queryLogFile,omitempty"`
+	// remoteWrite defines the remote write configuration, including URL, authentication, and relabeling settings.
+	// The remote-write protocol is intended to allow Prometheus instances to actively send the metrics they collect/receive to other instances.
+	// For more information, see: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write
+	// remoteWrite supports a maximum of 10 items in the list.
+	// +kubebuilder:validation:MaxItems=10
+	// +listType=map
+	// +listMapKey=url
+	// +optional
+	RemoteWrite []RemoteWriteSpec `json:"remoteWrite,omitempty"`
+	// resources defines the compute resource requests and limits for the Prometheus container.
+	// This includes CPU, memory and HugePages constraints to help control scheduling and resource usage.
+	// When not specified, defaults are used by the platform. Requests cannot exceed limits.
+	// This field is optional.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// This is a simplified API that maps to Kubernetes ResourceRequirements.
+	// The current default values are:
+	//   resources:
+	//    - name: cpu
+	//      request: 4m
+	//      limit: null
+	//    - name: memory
+	//      request: 40Mi
+	//      limit: null
+	// Maximum length for this list is 10.
+	// Minimum length for this list is 1.
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=10
+	// +kubebuilder:validation:MinItems=1
+	Resources []ContainerResource `json:"resources,omitempty"`
+	// retention defines the duration for which Prometheus retains data.
+	// This definition must be specified using the following regular
+	// expression pattern: `[0-9]+(ms|s|m|h|d|w|y)` (ms = milliseconds,
+	// s= seconds,m = minutes, h = hours, d = days, w = weeks, y = years).
+	// When omitted, this means the user has no opinion and the platform is left
+	// to choose reasonable defaults, which are subject to change over time.
+	// The default value is `15d`.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=20
+	// +optional
+	Retention string `json:"retention,omitempty"`
+	// retentionSize specifies the maximum volume of persistent storage that Prometheus uses for data blocks and the write-ahead log (WAL).
+	// Acceptable values use standard Kubernetes resource quantity formats, such as `Mi`, `Gi`, `Ti`, etc.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The default is no storage size limit is enforced and Prometheus will use the available storage capacity of the PersistentVolume.
+	// +kubebuilder:validation:MaxLength=20
+	// +optional
+	RetentionSize *string `json:"retentionSize,omitempty"`
```

Comment:
Similar comment, choose a unit to represent this.

Additionally, should this and the time of retention be co-located under a common parent field named `retention`?

Instead of:
```yaml
retention: ...
retentionSize: ...
```

you could do something like:
```yaml
retention:
  sizeInMi: 5000
  durationInSeconds: 3600
```

Reply:
Restructured per K8s API conventions - grouped into a single retention struct with explicit units:
```
retention:  durationInDays: 30    # 1-365 days
sizeInGiB: 100        # 1-16384 GiB
```
Both fields are optional with sensible defaults (15 days, no size limit). Using days instead of minutes since Prometheus retention is typically measured in days/weeks, and GiB for size since that's the practical unit for metrics storage. CMO will convert these to Prometheus format. @simonpasquier thoughts about this?